core/src/main/java/com/linecorp/armeria/server/HttpServerUpgradeHandler.java

```diff
@@ -75,6 +79,12 @@ final class HttpServerUpgradeHandler extends ChannelInboundHandlerAdapter {
 
     private static final FullHttpResponse UPGRADE_RESPONSE = newUpgradeResponse();
 
+    private static final ByteBuf INVALID_SETTINGS_HEADER_MESSAGE =
+            Unpooled.directBuffer().writeBytes(
+                    "Invalid HTTP2-Settings header\n".getBytes(StandardCharsets.UTF_8));
+    private static final FullHttpResponse INVALID_SETTINGS_HEADER_RESPONSE =
+            newInvalidSettingsHeaderResponse();
```

Comment:
This should not be cached, because otherwise we'll get `InvalidRefCntException`.